WalletWasabi/Services/MemPoolService.cs

```diff
@@ -120,9 +120,8 @@ public async Task<bool> TryPerformMempoolCleanupAsync(Func<Uri> destAction, IPEn
 				{
 					var compactness = 10;
 					var allMempoolHashes = await client.GetMempoolHashesAsync(compactness);
-					var allMempoolHashesHashSet = allMempoolHashes.ToHashSet();
 
-					var toRemove = TransactionHashes.Where(x => !allMempoolHashesHashSet.Contains(x.ToString().Substring(0, compactness)));
+					var toRemove = TransactionHashes.Where(x => !allMempoolHashes.Contains(x.ToString().Substring(0, compactness)));
```

Comment:
I needed to debug it in order to see what `ISet` type it returns and it returns a `HashSet`. That would means that the `Contains()` method that is called is really the `HashSet<string>.Contains()` and not the `ICollection<string>.Contains` as the intellisense says. The only problem imo is that it is not evident and is based on the fact that the json deserializer will return the concrete `HashSet`. I would let the `.ToHashSet`.

Anyway, I will run this and measure it as before just to be sure it doesn't revert the improvement achieved.